test/platform.test.ts

```diff
@@ -3,119 +3,596 @@
+                    displayName: "Windows PowerShell (x64)",
+                },
+                {
+                    exePath: "C:\\WINDOWS\\SysWOW64\\WindowsPowerShell\\v1.0\\powershell.exe",
+                    displayName: "Windows PowerShell (x86)",
+                },
+            ],
+            filesystem: {
+                "C:\\Program Files\\PowerShell": {
+                    "6": {
+                        "pwsh.exe": "",
                     },
-                    {
-                        versionName: platform.WindowsPowerShell32BitLabel,
-                        exePath: platform.SysWow64PowerShellPath,
+                    "7-preview": {
+                        "pwsh.exe": "",
                     },
-                ]);
-
-            checkFixedWindowsPowerShellpath(
-                platformDetails,
-                platform.SysnativePowerShellPath,
-                platform.System32PowerShellPath);
-        });
-
-        suite("64-bit Windows, 32-bit VS Code", () => {
-            const platformDetails: platform.IPlatformDetails = {
+                },
+                "C:\\Program Files (x86)\\PowerShell": {
+                    "6": {
+                        "pwsh.exe": "",
+                    },
+                    "7-preview": {
+                        "pwsh.exe": "",
+                    },
+                },
+                [pwshMsixDir]: {
+                    "pwsh.exe": "",
+                },
+                "C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0": {
+                    "powershell.exe": "",
+                },
+                "C:\\WINDOWS\\SysWOW64\\WindowsPowerShell\\v1.0": {
+                    "powershell.exe": "",
+                },
+            },
+        },
+        {
+            name: "Windows 64-bit, 64-bit VSCode (only Windows PowerShell)",
+            platformDetails: {
+                operatingSystem: platform.OperatingSystem.Windows,
+                isOS64Bit: true,
+                isProcess64Bit: true,
+            },
+            environmentVars: {
+                "ProgramFiles": "C:\\Program Files",
+                "ProgramFiles(x86)": "C:\\Program Files (x86)",
+                "windir": "C:\\WINDOWS",
+            },
+            expectedPowerShellSequence: [
+                {
+                    exePath: "C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
+                    displayName: "Windows PowerShell (x64)",
+                },
+                {
+                    exePath: "C:\\WINDOWS\\SysWOW64\\WindowsPowerShell\\v1.0\\powershell.exe",
+                    displayName: "Windows PowerShell (x86)",
+                },
+            ],
+            filesystem: {
+                "C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0": {
+                    "powershell.exe": "",
+                },
+                "C:\\WINDOWS\\SysWOW64\\WindowsPowerShell\\v1.0": {
+                    "powershell.exe": "",
```

Comment:
why the hashtable instead of just `path: exe`

Reply:
That's what `mock-fs` expects; this is their format.